client/landing/jetpack-cloud/components/sidebar/index.jsx

```diff
@@ -0,0 +1,208 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import Sidebar from 'layout/sidebar';
+import SidebarFooter from 'layout/sidebar/footer';
+import SidebarItem from 'layout/sidebar/item';
+import SidebarMenu from 'layout/sidebar/menu';
+import SidebarRegion from 'layout/sidebar/region';
+import ExpandableSidebarMenu from 'layout/sidebar/expandable';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+class JetpackCloudSidebar extends Component {
+	static propTypes = {
+		context: PropTypes.shape( {
+			path: PropTypes.string,
+		} ),
+	};
+
+	state = {
+		expandedSections: [],
+	};
+
+	componentDidMount() {
+		const section = this.getSectionFromCurrentPath();
+
+		if ( section ) {
+			this.toggleSection( section );
+		}
+	}
+
+	/**
+	 * Get current section name from path. When there's no section, we're in the dashboard.
+	 *
+	 * @returns {string} Current section
+	 */
+	getSectionFromCurrentPath() {
+		// If no section is present in the path, return 'dashboard'
+		return this.props.context.path.split( '/' )[ 2 ] ?? 'dashboard';
+	}
+
+	/**
+	 * Get current subsection name from path.
+	 *
+	 * @returns {string|null} Current subsection or null
+	 */
+	getSubsectionFromCurrentPath() {
+		return this.props.context.path.split( '/' )[ 3 ] ?? null;
+	}
+
+	/**
+	 * Toggle an expandable section.
+	 *
+	 * @param   {string} section Section to be toggled
+	 * @returns {void}
+	 */
+	toggleSection( section ) {
+		if ( this.isExpanded( section ) ) {
+			this.setState( {
+				expandedSections: this.state.expandedSections.filter( item => item !== section ),
+			} );
+		} else {
+			this.setState( {
+				expandedSections: [ ...this.state.expandedSections, section ],
+			} );
+		}
+	}
+
+	/**
+	 * Check if a section menu is expanded.
+	 *
+	 * @param   {string} section Section name
+	 * @returns {boolean}        Is the section expanded
+	 */
+	isExpanded( section ) {
+		return this.state.expandedSections.includes( section );
+	}
+
+	/**
+	 * Check if a menu item is selected based on the current path.
+	 *
+	 * @param   {string}  path Menu item path
+	 * @returns {boolean}      Is the menu item selected
+	 */
+	isSelected( path ) {
+		const section = this.getSectionFromCurrentPath();
+		const subsection = this.getSubsectionFromCurrentPath();
+
+		if ( ! subsection ) {
+			return section === path;
+		}
+
+		return section.concat( '/', subsection ) === path;
+	}
+
+	handleExpandableMenuClick( section ) {
+		return () => this.toggleSection( section );
+	}
+
+	onNavigate = () => {
+		window.scrollTo( 0, 0 );
+	};
+
+	render() {
+		return (
+			<Sidebar>
+				<SidebarRegion>
+					{ /* @todo: A profile info box needs to be created and added here; similar to <ProfileGravatar /> in client/me/sidebar/index.jsx */ }
+					<SidebarMenu>
+						<SidebarItem
```

Comment:
What do you think about dynamically declaring these items in an object or array and then looping through them here? It could save some of the boilerplate code.

Reply:
While I see *some* benefits of such approach, I think it's better as is. The reasons are:
* We'll remove some boilerplate code here but create boilerplate elsewhere (where the object is defined)
* In the new object we'll have to repeat most of the properties anyway like: `link`, `label`, `materialIcon` or `materialIconStyle`
* We'll add up unnecessary indirection
* In case we'll have to add some custom items here and there, our abstraction will become even more convoluted
* The current approach is used in other locations in the codebase where sidebar items are rendered

I'd stick to the simple implementation we have right now.

Reply:
Scratch that 🙂I've tested out the suggested approach and it turns out it's not that convoluted. We can also use the same structure for getting all top-level sections etc. This all has been addressed in 603c49e.